devlab/tests/generate_load.py

```diff
@@ -75,43 +75,68 @@ class NotFound(Exception):
 
 class BasePrerequisites(object):
 
-    def __init__(self, config, cloud_prefix='SRC'):
+    def __init__(self, config,
+                 configuration_ini,
+                 cloud_prefix='SRC'):
         self.filtering_utils = FilteringUtils()
         self.config = config
-        self.username = os.environ['%s_OS_USERNAME' % cloud_prefix]
-        self.password = os.environ['%s_OS_PASSWORD' % cloud_prefix]
-        self.tenant = os.environ['%s_OS_TENANT_NAME' % cloud_prefix]
-        self.auth_url = os.environ['%s_OS_AUTH_URL' % cloud_prefix]
-        self.image_endpoint = os.environ['%s_OS_IMAGE_ENDPOINT' % cloud_prefix]
-        self.neutron_endpoint = os.environ['%s_OS_NEUTRON_ENDPOINT'
-                                           % cloud_prefix]
+
+        self.cloud_prefix = cloud_prefix.lower()
+        self.configuration_ini = configuration_ini
+
+        if isinstance(configuration_ini, dict):
```

Comment:
Method arguments must either be of the same type, or follow the same interface, code should not check types of objects it works with.